system/jlib/jmetrics.cpp

```diff
@@ -140,52 +154,69 @@ bool MetricsManager::addMetric(const std::shared_ptr<IMetric> &pMetric)
     std::string name = pMetric->queryName();
 
     // Ensure metric name follows naming convention
-    if (std::regex_match(name, nameValidator))
+    try
     {
-        auto metaData = pMetric->queryMetaData();
-        std::unique_lock<std::mutex> lock(metricVectorMutex);
-        for (auto &metaDataIt: metaData)
+        if (std::regex_match(name, nameValidator))
```

Comment:
I don't think this function ever throws an exception - so I don't think it needs protecting.

Reply:
According the the header in regex.h, it throws a regex_error. I based the need for a try catch on this. If this is in error, I can remove it.

  /**
   * @brief Indicates if there is a match between the regular expression @p e
   * and a string.
   *
   * @param __s     [IN] The string to match.
   * @param __re    [IN] The regular expression.
   * @param __flags [IN] Controls how the regular expression is matched.
   *
   * @retval true  A match exists.
   * @retval false Otherwise.
   *
   * @throws an exception of type regex_error.
   */
  template<typename _Ch_traits, typename _Str_allocator,
	   typename _Ch_type, typename _Rx_traits>
    inline bool
    regex_match(const basic_string<_Ch_type, _Ch_traits, _Str_allocator>& __s,
		const basic_regex<_Ch_type, _Rx_traits>& __re,
		regex_constants::match_flag_type __flags
		= regex_constants::match_default)
    { return regex_match(__s.begin(), __s.end(), __re, __flags); }